c_binding/libsg.c

```diff
@@ -297,6 +297,13 @@ int _sg_io_vpd(char *err_msg, int fd, uint8_t page_code, uint8_t *data)
                                  sense_err_msg);
             }
         }
+        /* NVMe disk does not support SCSI VPD page 0x00 which fallback to here
+         */
+        if (ioctl_errno == ENOTTY) {
```

Comment:
Yeah. Normally, if a VPD query failed, libsg will check VPD page 0x00 to confirm that page is not supported. Then we check sense data. Apparently, NVMe does not support VPD page 0x00 or SCSI sense data, hence we failback to IOCTL error code.

RHEL 7 has some patch applied to kernel to support SCSI VPD pages for NVMe, but it does not works in Fedora kernel.